src/DictVectors/pdvec.jl

```diff
@@ -0,0 +1,797 @@
+    return 1, true
+end
+
+###
+### getting and setting
+###
+function Base.getindex(t::PDVec{K,V}, k::K) where {K,V}
+    segment_id, is_local = target_segment(t, k)
+    if is_local
+        return get(t.segments[segment_id], k, zero(V))
+    else
+        throw(CommunicatorError("Attempted to access non-local key `$k`"))
+    end
+end
+function Base.setindex!(t::PDVec{K,V}, val, k::K) where {K,V}
+    v = V(val)
+    segment_id, is_local = target_segment(t, k)
+    # Adding a key that is not local is supported. This is done to allow easy construction
+    # of vectors even when using MPI.
+    if is_local
+        if iszero(v)
+            delete!(t.segments[segment_id], k)
+        else
+            t.segments[segment_id][k] = v
+        end
+    end
+    return v
+end
+function deposit!(t::PDVec{K,V}, k::K, val, parent=nothing) where {K,V}
+    iszero(val) && return nothing
+    segment_id, is_local = target_segment(t, k)
+    if is_local
+        segment = t.segments[segment_id]
+        new_val = get(segment, k, zero(V)) + V(val)
+        if iszero(new_val)
+            delete!(segment, k)
+        else
+            segment[k] = new_val
+        end
+    end
+    return nothing
+end
+
+function Base.delete!(t::PDVec{K,V}, k::K) where {K,V}
+    t[k] = zero(V)
+    return t
+end
+
+###
+### empty(!), similar, copy, etc.
+###
+function Base.empty(
+    t::PDVec{K,V,N}; style=t.style, initiator=t.initiator, communicator=t.communicator,
+) where {K,V,N}
+    return PDVec{K,V,N}(; style, initiator, communicator)
+end
+function Base.empty(t::PDVec{K,V}, ::Type{V}; kwargs...) where {K,V}
+    return empty(t; kwargs...)
+end
+function Base.empty(t::PDVec{K,<:Any,N}, ::Type{V}; kwargs...) where {K,V,N}
+    return PDVec{K,V,N}(; kwargs...)
+end
+function Base.empty(t::PDVec{<:Any,<:Any,N}, ::Type{K}, ::Type{V}; kwargs...) where {K,V,N}
+    return PDVec{K,V,N}(; kwargs...)
+end
+function Base.empty!(t::PDVec)
+    Folds.foreach(empty!, t.segments, )
+    return t
+end
+Base.similar(t::PDVec, args...; kwargs...) = empty(t, args...; kwargs...)
+
+function Base.sizehint!(t::PDVec, n)
+    n_per_segment = cld(n, length(t.segments))
+    Folds.foreach(d -> sizehint!(d, n_per_segment), t.segments)
+    return t
+end
+
+function Base.copyto!(dst::PDVec, src::PDVec)
+    check_compatibility(dst, src)
+    Folds.foreach(dst.segments, src.segments) do d_seg, s_seg
+        copy!(d_seg, s_seg)
+    end
+    return dst
+end
+function Base.copy!(dst::PDVec, src::PDVec)
+    return copyto!(dst, src)
+end
+function Base.copy(src::PDVec)
+    return copy!(empty(src), src)
+end
+
+function localpart(t::PDVec{K,V,N,S,I}) where {K,V,S,I,N}
+    return PDVec{K,V,N,S,I,LocalPart}(
+        t.segments, t.style, t.initiator, LocalPart(t.communicator),
+    )
+end
+function localpart(t::PDVec{<:Any,<:Any,<:Any,<:Any,<:Any,<:LocalPart})
+    return t
+end
+
+###
+### Iterators, map, mapreduce
+###
+"""
+    PDVecKeys
```

Comment:
This docstring is not found when asking for it in the REPL
```
>? PDVecKeys
```
Is there a way to let it be found?

Reply:
Looks like aliases need separate docstrings. Added.